nixos/modules/security/acme.nix

```diff
@@ -220,10 +220,12 @@ let
         cat cert.pem chain.pem > fullchain.pem
         cat key.pem fullchain.pem > full.pem
 
-        chmod 640 *
-
         # Group might change between runs, re-apply it
         chown 'acme:${data.group}' *
+
+        # Default permissions make the files unreadable by group + anon
+        # Need to be readable by group
+        chmod 640 *
```

Comment:
0027 is already 640/750, by "Default permissions" you mean lego restricts the files even further?

Reply:
Yes, it does. I have a bit of discussion on it here: https://github.com/NixOS/nixpkgs/issues/121293#issuecomment-830354176 with source links.